app/assets/javascripts/discourse/app/components/topic-dismiss-buttons.js

```diff
@@ -0,0 +1,104 @@
+import { action } from "@ember/object";
+import showModal from "discourse/lib/show-modal";
+import { later } from "@ember/runloop";
+import isElementInViewport from "discourse/lib/is-element-in-viewport";
+import discourseComputed, { on } from "discourse-common/utils/decorators";
+import I18n from "I18n";
+import Component from "@ember/component";
+
+export default Component.extend({
+  tagName: "",
+  classNames: ["topic-dismiss-buttons"],
+
+  position: null,
+  selectedTopics: null,
+  model: null,
+
+  @discourseComputed("position")
+  containerClass(position) {
+    return `dismiss-container-${position}`;
+  },
+
+  @discourseComputed("position")
+  dismissReadId(position) {
+    return `dismiss-topics-${position}`;
+  },
+
+  @discourseComputed("position")
+  dismissNewId(position) {
+    return `dismiss-new-${position}`;
+  },
+
+  @discourseComputed(
+    "position",
+    "isOtherDismissUnreadButtonVisible",
+    "isOtherDismissNewButtonVisible"
+  )
+  showBasedOnPosition(
+    position,
+    isOtherDismissUnreadButtonVisible,
+    isOtherDismissNewButtonVisible
+  ) {
+    let positionShouldShow =
+      position === "top"
+        ? !(isOtherDismissUnreadButtonVisible || isOtherDismissNewButtonVisible)
+        : true;
+
+    return positionShouldShow;
+  },
+
+  @discourseComputed("selectedTopics.length")
+  dismissLabel(selectedTopicCount) {
+    if (selectedTopicCount === 0) {
+      return I18n.t("topics.bulk.dismiss_button");
+    }
+    return I18n.t("topics.bulk.dismiss_button_with_selected", {
+      count: selectedTopicCount,
+    });
+  },
+
+  @discourseComputed("selectedTopics.length")
+  dismissNewLabel(selectedTopicCount) {
+    if (selectedTopicCount === 0) {
+      return I18n.t("topics.bulk.dismiss_new");
+    }
+    return I18n.t("topics.bulk.dismiss_new_with_selected", {
+      count: selectedTopicCount,
+    });
+  },
+
+  // we want to only render the Dismiss... button at the top of the
+  // page if the user cannot see the bottom Dismiss... button based on their
+  // viewport, or if too many topics fill the page
+  @on("didInsertElement")
+  _determineOtherDismissVisibility() {
+    later(() => {
+      if (this.position === "top") {
+        this.set(
+          "isOtherDismissUnreadButtonVisible",
+          isElementInViewport(document.getElementById("dismiss-topics-bottom"))
+        );
+        this.set(
+          "isOtherDismissNewButtonVisible",
+          isElementInViewport(document.getElementById("dismiss-new-bottom"))
+        );
+      } else {
+        this.set("isOtherDismissUnreadButtonVisible", true);
+        this.set("isOtherDismissNewButtonVisible", true);
+      }
+    });
```

Comment:
```suggestion
  @afterRender
  _determineOtherDismissVisibility() {
    if (this.position === "top") {
      this.set(
        "isOtherDismissUnreadButtonVisible",
        isElementInViewport(document.getElementById("dismiss-topics-bottom"))
      );
      this.set(
        "isOtherDismissNewButtonVisible",
        isElementInViewport(document.getElementById("dismiss-new-bottom"))
      );
    } else {
      this.set("isOtherDismissUnreadButtonVisible", true);
      this.set("isOtherDismissNewButtonVisible", true);
    }
```

(needs: `import { afterRender } from "discourse-common/utils/decorators";`)

Reply:
@CvX afterRender did not work for me at all here for some reason, the code inside the afterRender function was never executed?

Reply:
🤦 Gah, my bad! I always forget how `@afterRender` works! It's not "run this code after rendering" but "when called, delay the execution after rendering". So the correct suggestion would be either:

```js
  didInsertElement() {
    _super.didInsertElement(...arguments);

    // code
```

or:

```js
  init() {
    this._super(...arguments);
    this._determineOtherDismissVisibility();
  }

  @afterRender
  _determineOtherDismissVisibility() {
    // code
```

Reply:
Cool, will just leave it as is then :)

Reply:
Is the `later()` call required though?

Reply:
As far as I could tell, yes. If I removed it I did not get the desired result — the other element is not counted as visible even though it clearly is.